src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/timeserved/TimeServedProbationConfirmController.kt

```diff
@@ -0,0 +1,77 @@
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.request.timeserved.TimeServedProbationConfirmContactRequest
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.resource.Tags
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.timeserved.TimeServedProbationConfirmContactService
+
+@Tag(name = Tags.TIME_SERVED_PROBATION_CONFIRM_CONTROLLER)
+@RestController
+@RequestMapping("/licences/time-served")
+class TimeServedProbationConfirmController(
+  private val timeServedProbationConfirmContactService: TimeServedProbationConfirmContactService,
+) {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(TimeServedProbationConfirmController::class.java)
+  }
+
+  @PutMapping("{licenceId}/confirm/probation-contact")
+  @PreAuthorize("hasAnyRole('CVL_ADMIN')")
+  @Operation(
+    summary = "Adds the probation contact confirmation for a time served licence.",
+    description = "Adds the probation contact status and communication methods for a time served licence. Requires ROLE_CVL_ADMIN.",
+    security = [SecurityRequirement(name = "ROLE_CVL_ADMIN")],
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(responseCode = "204", description = "Probation contact confirmation added successfully"),
+      ApiResponse(
+        responseCode = "400",
+        description = "Bad request — request body must be valid",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorised — requires a valid OAuth2 token",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden — requires an appropriate role",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Licence record not found",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  fun addTimeServedProbationConfirmContact(
+    @PathVariable("licenceId") licenceId: Long,
+    @Valid @RequestBody body: TimeServedProbationConfirmContactRequest,
+  ) {
+    log.info("Adding probation contact confirmation for licenceId=$licenceId with body=$body")
+    timeServedProbationConfirmContactService.addConfirmContact(licenceId, body)
+    log.debug("Probation contact confirmation added successfully for licenceId=$licenceId")
+  }
```

Comment:
I agree with the PR comment and think we could merge this into the time served controller - it only has `update` and `get` now

maybe in a follow up if you're repackaging things?